lib/options.nix

```diff
@@ -333,5 +333,45 @@ rec {
         else
           example;
     };
+
+  mkLazyLoadOption =
+    originalName:
+    lib.mkOption {
+      description = ''
+        Lazy-load settings for ${originalName}.
+      '';
+      default = { };
+      type = types.submodule (
+        { config, ... }:
+        {
+          options = {
+            enable = lib.mkOption {
+              default = lib.any (x: x != null) (builtins.attrValues config.settings);
```

Comment:
This should have a `literalExpression` or `literalMD` type defaultText. Maybe something like:

```nix
defaultText = literalMD ''
  `true` when `settings` has a non-null attribute
'';
```

----

But while we're discussing the option:

Is this actually what we want long-term, or would we rather have most/all plugins default to `lazy.enable = true` but have it _also_ require a global `lazy.enable`?

If we did go with a global enable, should it be an explicit (user-set) option, or should it just be an internal option based on having a lazyload impl plugin enabled?

Reply:
I think it makes more sense, for now. I think we'd need to update plugins with sane defaults of cmd/ft/etc triggers to support enabling by default without user configuration.

Reply:
Agreed. I hadn't fully thought this one through. Of course we can't lazy load plugins that have no "trigger" events configured.